src/mpsc/async_impl.rs

```diff
@@ -422,6 +422,76 @@ feature! {
             }
         }
 
+        /// Attempts to receive the next message for this receiver by reference
+        /// without blocking.
```

Comment:
for the async MPSC, we should probably not use the term "blocking"; the async channel will never block, but it might _wait_:
```suggestion
        /// without waiting for a new message if the channel is empty..
```